lib/google/gax/operation.rb

```diff
@@ -0,0 +1,138 @@
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# rubocop:disable Style/Documentation
+# Without this, somehow rubocop raises an warning for documentaion for
+# module Gax, however it's documented in lib/google/gax.rb.
+
+# This must be loaded separate from google/gax to avoid circular dependency.
+require 'google/gax/constants'
+require 'google/gax/settings'
+require 'google/longrunning/operations_api'
+
+module Google
+  module Gax
+    # A class used to wrap Google::Longrunning::Operation objects.
+    #
+    # @example General usage example
+    #   require 'google/gax/operation'
+    #
+    #   op = Google::Gax::Operation.new api.methodThatReturnsOperation(name)
+    #   op.done?
+    #   op.reload!
+    #   op.done?
+    #   results = op.results
+    #
+    # @attribute [r] grpc_op
+    #   @return [Google::Longrunning::Operation] The wrapped grpc
+    #     operation object.
+    # @attribute [r] client
+    #   @return [Google::Longrunning::OperationsApi] The client that handles the
+    #     grpc operations.
+    class Operation
+      attr_reader :grpc_op, :client
+
+      # @param grpc_op [Google::Longrunning::Operation]
+      #   The inital longrunning operation.
+      # @param client [Google::Longrunning::OperationsApi]
+      #   The client that handles the grpc operations.
+      def initialize(grpc_op, client: nil)
+        @grpc_op = grpc_op
+        @client = client.nil? Google::Longrunning::OperationsApi.new : client
+      end
+
+      # If the operation is done, returns the result, otherwise returns nil.
+      #
+      # @return [nil | Google::Rpc::Status | Object]
+      #   The result of the operation
+      def results
+        return nil unless done?
+        return @grpc_op.error if @grpc_op.result == :error
+        @grpc_op.response.value if @grpc_op.result == :response
```

Comment:
`response` is actually `google.protobuf.Any`. Does protobuf-ruby automatically unpack the result? otherwise, `Google::Protobuf::Any.unpack` might be required.